web/static/js/components/surveys/SurveyShow.jsx

```diff
@@ -51,6 +51,30 @@ class SurveyShow extends Component {
       return <p>Loading...</p>
     }
 
+    let primaryMode = <div className='mode'>
+      <span className='type'>Primary Mode</span>
+      <div>
+        <i className='material-icons'>{survey.mode[0] == 'sms' ? 'sms' : 'phone'}</i>
```

Comment:
Extract the right hand side of `let primaryMode = ...` to a function that receives `survey.mode`.